src/installer/managed/Microsoft.NET.HostModel/Bundle/TargetInfo.cs

```diff
@@ -57,17 +57,17 @@ public TargetInfo(OSPlatform? os, Architecture? arch, Version targetFrameworkVer
                 throw new ArgumentException($"Invalid input: Unsupported Target Framework Version {targetFrameworkVersion}");
             }
 
-            if (IsLinux && Arch == Architecture.Arm64)
+            if (IsWindows)
             {
-                // We align assemblies in the bundle at 4K so that we can use mmap on Linux without changing the page alignment of ARM64 R2R code.
+                // We align assemblies in the bundle at 4K - per requirements of memory mapping API (MapViewOfFile3, et al).
                 // This is only necessary for R2R assemblies, but we do it for all assemblies for simplicity.
-                // See https://github.com/dotnet/runtime/issues/41832.
                 AssemblyAlignment = 4096;
             }
-            else if (IsWindows)
+            else if (Arch == Architecture.Arm64)
             {
-                // We align assemblies in the bundle at 4K - per requirements of memory mapping API (MapViewOfFile3, et al).
+                // We align assemblies in the bundle at 4K so that we can use mmap on Unix without changing the page alignment of ARM64 R2R code.
                 // This is only necessary for R2R assemblies, but we do it for all assemblies for simplicity.
+                // See https://github.com/dotnet/runtime/issues/41832.
```

Comment:
Once you get past the assert, you may still see crashes. I think they were because of this. We align files in the bundle on 4K for Linux, to make sure that `adrp` assembly instructions (which works off a 4K offset) are stable without performing relocations. 

I think the same applies to OSX as well. The fix could be older than Apple Silicon and therefore covered Linux only.

Reply:
This is in a compile-time tool. It cold be a completely different platform/os.
We do not want arbitrary big alignment too, since it will increase the total size of the single file app.

Reply:
Also this is because of `ADRP` instruction, which operates with 4k offsets regardless of OS page.